pkgs/development/libraries/cosmopolitan/default.nix

```
@@ -0,0 +1,45 @@
+{ lib, stdenv, makeWrapper, runCommand, cosmopolitan }:
+stdenv.mkDerivation {
+  pname = "cosmopolitan-unstable";
+  version = "20210228";
+  src = builtins.fetchGit {
+    url = "https://github.com/jart/cosmopolitan";
+    rev = "94afa982c35d4322999b5591f6d41d5cfc3f19a5";
+    ref = "master";
+  };
+  postPatch = ''
+    sed -i -e "s|/usr/bin/env bash|/bin/sh|" build/findtmp
+  '';
+  dontConfigure = true;
+  dontFixup = true;
+  nativeBuildInputs = [ makeWrapper ];
+  enableParallelBuilding = true;
+  preBuild = ''
+    makeFlagsArray=(SHELL=/bin/sh "MKDIR=mkdir -p")
+  '';
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,lib/include}
+    install o/cosmopolitan.h $out/lib/include
+    install o/cosmopolitan.a o/libc/crt/crt.o o/ape/ape.{o,lds} $out/lib
+    makeWrapper ${stdenv.cc}/bin/gcc $out/bin/cosmoc --add-flags "-O -static -nostdlib -nostdinc -fno-pie -no-pie -mno-red-zone -fuse-ld=bfd -Wl,-T,$out/lib/ape.lds -include $out/lib/include/cosmopolitan.h $out/lib/{crt.o,ape.o,cosmopolitan.a}"
+    runHook postInstall
+  '';
+  passthru.tests = {
+    hello = runCommand "hello-world" { } ''
+      printf 'main() { printf("hello world\\n"); }\n' >hello.c
+      ${stdenv.cc}/bin/gcc -g -O -static -nostdlib -nostdinc -fno-pie -no-pie -mno-red-zone -o hello.com.dbg hello.c \
+        -fuse-ld=bfd -Wl,-T,${cosmopolitan}/lib/ape.lds -include ${cosmopolitan}/lib/include/cosmopolitan.h ${cosmopolitan}/lib/crt.o ${cosmopolitan}/lib/ape.o ${cosmopolitan}/lib/cosmopolitan.a
+      ${stdenv.cc.bintools.bintools_bin}/bin/objcopy -S -O binary hello.com.dbg hello.com
+      ./hello.com
+      printf "test successful" > $out
+    '';
+  };
+  meta = with lib; {
+    homepage = "https://justine.lol/cosmopolitan/";
+    description = "your build-once run-anywhere c library";
+    platforms = platforms.all;
```

```suggestion
    platforms = platforms.x86_64;
```
checked on my RPi4, cosmopolitan currently only supports intel64. Maybe in future revisions idk.

---

"It'll be nice to know that any normal PC program we write will "just work" on Raspberry Pi and Apple ARM. All we have to do embed an ARM build of the emulator above within our x86 executables, and have them morph and re-exec appropriately, similar to how Cosmopolitan is already doing doing with qemu-x86_64, except that this wouldn't need to be installed beforehand."  Sounds like it can happen. :crossed_fingers: